msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationJavaTest.java

```diff
@@ -406,11 +409,14 @@ public void onError(@NonNull BaseException exception) {
         application.signIn(username, null, signInCallback);
         // 1a. Server returns invalid user error
         assertTrue(signInResult.get(30, TimeUnit.SECONDS) instanceof SignInResult.CodeRequired);
-        SignInCodeRequiredState nextState = (((SignInResult.CodeRequired) signInResult.get(30, TimeUnit.SECONDS)).getNextState());
+        SignInCodeRequiredState nextState = spy((((SignInResult.CodeRequired) signInResult.get(30, TimeUnit.SECONDS)).getNextState()));
```

Comment:
Why we would use spy() here? Any particular advantage? It's a confusion not a suggestion, feel free to revolt the comment.

Reply:
Good question. I'd recommend reading a bit about Mockito's `mock` vs. `spy`. In short: with spy you can wrap a real object into a partially mocked one. Any method/field that's not overwritten by the spy, will behave as normal. But the methods that are overwritten by the spy, will work as a mocked variant would. In this case it's helpful because, when using mock API, `correlationID` is going to be null. Using a spy we can change this field into some mocked value, so the flow continues to work as expected.